tests/integration/conftest.py

```diff
@@ -93,22 +110,77 @@ async def app(
     # mypy doesn't see that ActiveStatus has a name
     await model.wait_for_idle(status=ActiveStatus.name)  # type: ignore
 
+    # test that the application is online
+    for _ in range(10):
+        try:
+            response = requests.get(
+                f"http://{await get_mattermost_ip(ops_test, application)}:8065",
+                timeout=5
+            )
+            if response.status_code == 200:
+                break
+        except:
+            pass
+        # wait a bit
+        time.sleep(10)
+
```

Comment:
There's no guarantee that we will have a running app after the 10 tries and we still yield the 'application' object, correct?

Reply:
Yes, correct. If that happens, then the "workload_online" integration test will fail. This is done only to give a bit more time to mattermost to come online before this test.
If we want to guarantee that mattermost is reachable if the charm says it's active, then we need to fix it properly.

Reply:
I decided to use David's idea.